docs/Examples/get-tx-from-chain.js

```diff
@@ -1,47 +1,85 @@
 'use strict';
 
 const bcoin = require('../..');
+
 const Logger = require('blgr');
 
 // Setup logger to see what's Bcoin doing.
 const logger = new Logger({
   level: 'debug'
 });
 
-// Create chain for testnet, specify chain directory
+// Create an in-memory blockchain.
 const chain = new bcoin.Chain({
   logger: logger,
+  memory: true,
+  network: 'testnet'
+});
+
+const mempool = new bcoin.Mempool({ chain: chain });
+
+// Create a network pool of peers with a limit of 8 peers.
+const pool = new bcoin.Pool({
+  logger: logger,
+  chain: chain,
+  mempool: mempool,
+  maxPeers: 8
+});
+
+// Create a chain indexer which indexes tx by hash
+const indexer = new bcoin.TXIndexer({
+  logger: logger,
+  memory: true,
   network: 'testnet',
-  db: 'leveldb',
-  prefix: '/tmp/bcon-testnet',
-  indexTX: true,
-  indexAddress: true
+  chain: chain
 });
 
-(async () => {
+// Open the pool (implicitly opens mempool and chain).
+(async function() {
   await logger.open();
+
+  await pool.open();
+
+  // Connect, start retrieving and relaying txs
+  await pool.connect();
+
+  // Start the blockchain sync.
+  pool.startSync();
+
   await chain.open();
```

Comment:
if `pool.open()` implicitly opens the mempool and chain is `chain.open()` still necessary?

Reply:
Don't think `pool.open` implicity opens `mempool` or `chain`, maybe you're thinking of `fullnode`?

Reply:
line 37 says it does: https://github.com/bcoin-org/bcoin/pull/424/files/cd153cc94592f2b0d5f94d557c1682d6ce2a0c19#diff-1ee8911cae33bdcf90a90a90f5d7bdaeR37

Reply:
Ah, good catch, that is outdated. Updated now.